### PR TITLE
Fast implementation of iterate

### PR DESCRIPTION
**Pull request by ScottPJones**

### CODE REVIEW

**Comment by ScottPJones** [comment]
@xiaodaigh @oxinabox Could you all review this please?

**Review by xiaodaigh — CHANGES_REQUESTED** [review]
I think we need some test cases for the iterate.

File: src/base.jl [hunk 1]
```diff
@@ -49,14 +49,24 @@ Base.convert(::String, ss::ShortString) = String(ss)
...
+@inline _get_word(s::ShortString{T}, i::Int) where {T} =
+           (s.size_content >> (8*(sizeof(T) - i - 3)))%UInt32
+
+@inline function Base.iterate(s::ShortString, i::Int=1)
+       0 < i <= ncodeunits(s) || return nothing
+       chr = _get_word(s, i)
+       chr < 0x8000_0000 ? (reinterpret(Char, chr & 0xFF00_0000), i + 1) :
+           chr < 0xe000_0000 ? (reinterpret(Char, chr & 0xFFFF_0000), i + 2) :
+           chr < 0xf000_0000 ? (reinterpret(Char, chr & 0xFFFF_FF00), i + 3) :
+           (reinterpret(Char, chr), i + 4)
```
Comment (oxinabox) [inline src/base.jl:66]: Purely a style comment but i find for multiline `?` statmeents `if`, `elseif`, `elseif` is much easier to read.

**Comment by ScottPJones** [comment]
@xiaodaigh I added Lyndon's tests (using collect to test iterate)

**Pull request merged**